tools/scripts/docusaurus/generate_docs_params.go

```diff
@@ -1,183 +1,222 @@
+)
 
 func init() {
-	paramsTempalteFile, err := os.ReadFile("./tools/scripts/docusaurus/params_template.md")
+	flag.StringVar(&flagProtoRootPathValue, "proto-root", "./proto", "path to the proto files root directory; this directory will be walked, looking for Params.proto files")
+	flag.Parse()
+}
+
+func main() {
+	// Parse templates.
+	templs, err := template.ParseFiles(paramsDocTemplatePath, paramFieldRowTemplatePath)
 	if err != nil {
-		polylog.DefaultContextLogger.Error().Err(err).Send()
+		logger.Error().Err(err).Msgf("Unable to parse template path %q", paramsDocTemplatePath)
 		os.Exit(1)
 	}
-	paramsDocsTemplateStr = string(paramsTempalteFile)
-}
 
-// writeContentToFile writes the given content to the specified file.
-func writeContentToFile(file_path, content string) error {
-	file, err := os.Create(file_path)
+	// Interpolate templates.
+	docs, err := prepareGovernanceParamsDocs(flagProtoRootPathValue, templs)
 	if err != nil {
-		return err
+		logger.Error().Err(err).Msg("Error preparing governance params docs")
+		os.Exit(1)
 	}
-	defer file.Close()
 
-	// Write the string to the file
-	_, err = file.WriteString(content)
+	// Write output to destination.
+	err = writeContentToFile(destinationPath, docs)
 	if err != nil {
-		return err
+		logger.Error().Err(err).Msg("Error writing content to file")
+		os.Exit(1)
 	}
 
-	return nil
 }
 
-// findProtoFiles returns a slice of file paths that contain the specified pattern within the given base directory.
-func findProtoFiles(baseDir, pattern string) (protoFilePaths []string, err error) {
-	err = filepath.Walk(baseDir, func(path string, info os.FileInfo, err error) error {
-		if err != nil {
-			return err
-		}
+// prapareGovernanceParamsDocs recursively walks the filesystem starting from
+// protoFilesRootDir, looking for files names matching "params.proto". For each
+// matching proto file, the fields on any "Params" message types are discovered.
+// All discovered param fields are interpolated into the templates provided by
+// templs and the final output is returned.
+func prepareGovernanceParamsDocs(protoFilesRootDir string, templs *template.Template) (string, error) {
+	paramsDocOutputBuf := new(bytes.Buffer)
+	paramFieldRowsOutputBuf := new(bytes.Buffer)
+	paramsFieldNodesByModule := make(map[string][]*ast.FieldNode)
+
+	if pathWalkErr := filepath.Walk(
+		protoFilesRootDir,
+		forEachMatchingFileWalkFn(
+			"params.proto",
+			newCollectParamsFieldNodesFn(paramsFieldNodesByModule),
+		),
+	); pathWalkErr != nil {
+		logger.Error().Err(pathWalkErr)
+		os.Exit(1)
+	}
+
+	for moduleName, fieldNodes := range paramsFieldNodesByModule {
+		for _, fieldNode := range fieldNodes {
+			// Uncomment and concatenate the field's comment lines.
+			comment := ""
+			for commentIdx, commentLine := range fieldNode.LeadingComments() {
+				var commentFmt = " %s"
+				if commentIdx == 0 {
+					commentFmt = "%s"
+				}
+
+				comment += fmt.Sprintf(commentFmt, strings.Trim(commentLine.Text, " /"))
```

Comment:
Shouldn't we trim `//`?

Reply:
The second argument to `strings.Trim` is the "cutlist": a set of characters which are removed. It is not a substring match. Therefore, this trims ANY spaces AND forward slashed from the edges of the string.